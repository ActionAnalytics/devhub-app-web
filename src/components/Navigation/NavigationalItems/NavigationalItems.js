import React from 'react';
import NavigationalItem from './NavigationalItem/NavigationalItem';
import classes from './NavigationalItems.module.css';

const NavigationalItems = props => {
  const navItems = props.navItems.map((item, ind) => (
    <NavigationalItem
      key={`${item.text}_${ind}`}
      icon={item.icon}
      title={item.text}
      link={item.link}
    />
  ));

  return <div className={classes.NavigationalItems}>{navItems}</div>;
};

export default NavigationalItems;
