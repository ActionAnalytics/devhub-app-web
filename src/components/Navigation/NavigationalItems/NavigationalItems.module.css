.NavigationalItems {
    list-style: none;
    margin: 0;
    padding: 0;
    display: flex;
    flex-flow: row wrap;
    max-width: 800px;
    margin: 0 auto;
}