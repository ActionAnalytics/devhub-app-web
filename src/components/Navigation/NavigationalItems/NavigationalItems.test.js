import React from 'react';
import { shallow } from 'enzyme';
import NavigationalItems from './NavigationalItems';

describe('Navigational Item Component', () => {
  test('it matches snapshot', () => {
    const links = [{ text: 'link1', link: '/' }, { text: 'link2', link: '/2' }];
    const wrapper = shallow(<NavigationalItems navItems={links} />);
    expect(wrapper).toMatchSnapshot();
  });
});
