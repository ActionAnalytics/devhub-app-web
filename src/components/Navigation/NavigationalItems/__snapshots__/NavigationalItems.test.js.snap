// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigational Item Component it matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavigationalItems
    navItems={
      Array [
        "link 1",
        "link 2",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavigationalItem
          description=""
          icon="coffee"
          title="link 1"
        />,
        <NavigationalItem
          description=""
          icon="coffee"
          title="link 2"
        />,
      ],
      "className": "NavigationalItems",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "link 1_0",
        "nodeType": "function",
        "props": Object {
          "description": "",
          "icon": "coffee",
          "title": "link 1",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "link 2_1",
        "nodeType": "function",
        "props": Object {
          "description": "",
          "icon": "coffee",
          "title": "link 2",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavigationalItem
            description=""
            icon="coffee"
            title="link 1"
          />,
          <NavigationalItem
            description=""
            icon="coffee"
            title="link 2"
          />,
        ],
        "className": "NavigationalItems",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "link 1_0",
          "nodeType": "function",
          "props": Object {
            "description": "",
            "icon": "coffee",
            "title": "link 1",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "link 2_1",
          "nodeType": "function",
          "props": Object {
            "description": "",
            "icon": "coffee",
            "title": "link 2",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
